src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -98,12 +116,6 @@ protected JsonSerializerContext(JsonSerializerOptions? options)
 
         JsonTypeInfo? IJsonTypeInfoResolver.GetTypeInfo(Type type, JsonSerializerOptions options)
         {
```

Comment:
this logic was correct because we replace this interface implementation in the generated code. Also consider throwing for null options

Reply:
I changed it because the `JsonSerializerOptions` copy constructor will now also include any `JsonSerializerContext`, so if a user-derived `JsonSerializerContext` were copied then resolving metadata would always trigger this exception. Perhaps though it would still fail downstream because `jsonTypeInfo.Options` doesn't match?

Reply:
Actually scratch that, I'm keeping as-is. It should be the responsibility of `JsonSerializerOptions` to perform this validation, which it already does.

Reply:
can you clarify? as is = old behavior or as in PR? keeping this as in the PR would not be correct since user might end up using type info constructed for different options but true that we will currently validate that and throw if that's the case. IMO old behavior was correct

Reply:
> IMO old behavior was correct

I can revert it, but honestly I don't think it's important since it manifests in an unsupported scenario.